rest/route/github.go

```diff
@@ -763,7 +763,7 @@ func (gh *githubHookApi) createVersionForTag(ctx context.Context, pRef model.Pro
 			"tag":                tag,
 			"message":            "user not authorized for git tag version",
 		})
-		return nil, nil
+		return nil, errors.Errorf("user '%s' not authorized to create git tag versions for project '%s'", tag.Pusher, pRef.Id)
```

Comment:
So re your PR comment, I don't think we need to do a new subscription and etc -- I _think_ we could just send an email or something directly here by, assuming senders are set up (like here https://github.com/evergreen-ci/evergreen/blob/276cd4ec09474324c8fbd6ad7fcd9bf20c963d8a/rest/route/notification.go#L156-L162). 

I don't think we'd want it to be something users need to opt in to, because that doesn't really solve the problem.

I'm also fine with just going with the docs improvements. (In which case maybe we could also make a tiny spruce PR to make this text a little more explicit)

![image](https://github.com/evergreen-ci/evergreen/assets/26798134/1d88c671-9057-48d7-87b5-e497537f5553)

Reply:
Ah, my assumption was for sending notifications we always go the event --> check subscription --> send notification pipeline detailed [here](https://docs.google.com/document/d/1KxK2-JhKiHg7ydoEJN6rAf63k94KE_5-DqlYBj48pig/edit#heading=h.sffkah3u5q9), since I didn't see any examples in our codebase of us just sending notifications out arbitrarily.